### PR TITLE
Imodel02 pr validation

### PR DESCRIPTION
**Pull request by DanRod1999**
New PR after changing branch name.
Original PR: https://github.com/iTwin/itwinjs-core/pull/4278
Original Issue: https://github.com/iTwin/itwinjs-core/issues/4086

### CODE REVIEW

File: common/config/azure-pipelines/jobs/fast-ci.yaml [hunk 1]
```diff
@@ -27,12 +27,49 @@ pr:
...
+  timeoutInMinutes: 60
+
+  workspace:
+    clean: all
+
+  steps:
+  - checkout: self
+    clean: true
+  - template: ../templates/core-build.yaml
     parameters:
```
Comment (aruniverse) [inline common/config/azure-pipelines/jobs/fast-ci.yaml:68]: i think you need another param for buildIos like the old ci-core had

image

Comment (aruniverse) [reply common/config/azure-pipelines/jobs/fast-ci.yaml:68]: i dont think it should be always true, only when the agent is a mac. no xcode on windows or linux

Comment (DanRod1999) [reply common/config/azure-pipelines/jobs/fast-ci.yaml:68]: Ah ok. It was set to true in original fast-ci.yaml, but I understand now

File: common/config/azure-pipelines/jobs/regression-testing.yaml [hunk 2]
```diff
@@ -17,33 +17,109 @@ schedules:
...
+  variables:
+  - name: linuxPool
+    value: 'iModeljs Linux'
+  - name: mac_pool
+    value: iModelTechCI
+  - name: win_pool
+    value: iModelTechCI
+
+  strategy:
+    matrix:
```
Comment (calebmshafer) [inline common/config/azure-pipelines/jobs/regression-testing.yaml:30]: With adding all of these to the matrix, can you limit the number in parallel? I think before it's run each set of jobs based on Node versio with a max number of concurrent jobs of 3.

File: common/config/azure-pipelines/templates/core-build.yaml [hunk 3]
```diff
@@ -70,3 +99,10 @@ steps:
   - script: node common/scripts/install-run-rush.js docs
     displayName: rush docs
     workingDirectory: ${{ parameters.workingDir }}
+
+  # # Will run if even there is a failure somewhere else in the pipeline.
```
Comment (calebmshafer) [inline common/config/azure-pipelines/templates/core-build.yaml:103]: Why is this commented out?

**Review by calebmshafer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**